src/main/java/core/basesyntax/strategy/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        String key = transaction.getFruit();
+        Integer currentQuantity = Storage.storage.get(key);
```

Comment:
Here and in other operation handlers. Consider creating and using dao layer for data manipulations against storage.